reference/04-Patterns/block-bindings-api.md

```diff
@@ -0,0 +1,242 @@
+	}
+}
+```
+
+1. First we are looking for our keys.  If no key is set telling us how to map our dynamic data, we will return null in our callback. The pattern will then use the defaults set when the pattern is created.
+2. Next we create defaults for our keys.  These can be the same items set when creating our pattern.  Here for example, we will create our pattern with generic messaging leading to a Contact page and set our defaults to the same.
+3. We then get our data from our post meta fields.  If we have all desired data, we can then remap our bindings to dynmaic text to include our product's info.  If any if the data is missing, we have our defaults to fall back on.
+4. Finally, we return each key in a switch statement.
+
+All that's it!  Similar to the steps outlined in [Synced Patterns](../../reference/Patterns/synced-patterns#creating-synced-patterns-for-theme-authors), we can then create a pattern in our theme and manually set our blocks attributes as outlined above using our custom source's slug and our callback functions keys to sync it in our environments.
+
+Below is the full code from our example Call to Action.
+
+```html
+<!-- wp:group {"metadata":{"name":"Call to Action"},"style":{"spacing":{"blockGap":"var:preset|spacing|10","padding":{"top":"var:preset|spacing|20","bottom":"var:preset|spacing|20","left":"var:preset|spacing|20","right":"var:preset|spacing|20"}}},"backgroundColor":"accent-4","layout":{"type":"flex","orientation":"vertical","justifyContent":"center"}} -->
+<div class="wp-block-group has-accent-4-background-color has-background" style="padding-top:var(--wp--preset--spacing--20);padding-right:var(--wp--preset--spacing--20);padding-bottom:var(--wp--preset--spacing--20);padding-left:var(--wp--preset--spacing--20)">
+
+<!-- wp:heading {"metadata":{"bindings":{"content":{"source":"tenup/cta-binding","args":{"key":"cta_heading"}}}},"fontSize":"x-large"} -->
+<h2 class="wp-block-heading has-x-large-font-size">Want to get in touch?</h2>
+<!-- /wp:heading -->
+
+<!-- wp:paragraph {"metadata":{"bindings":{"content":{"source":"tenup/cta-binding","args":{"key":"cta_paragraph"}}}},"fontSize":"medium"} -->
+<p class="has-medium-font-size">We would love to hear from you!</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:buttons -->
+<div class="wp-block-buttons">
+
+<!-- wp:button {"metadata":{"bindings":{"url":{"source":"tenup/cta-binding","args":{"key":"cta_button_url"}},"text":{"source":"tenup/cta-binding","args":{"key":"cta_button_text"}}}}} -->
+<div class="wp-block-button">
+	<a class="wp-block-button__link wp-element-button" href="/contact-us">Contact Us</a>
+</div>
+<!-- /wp:button -->
+
+ </div>
+<!-- /wp:buttons -->
+
+ </div>
+<!-- /wp:group -->
+```
+
+Here we can see that our Call to Action will display a nice fallback should our post meta fields be missing.
+
+![Call to action with default settings applied](../../static/img/block-pattern-call-to-action-example.jpg)
+<br/>
+*Call to action with default settings applied*
```

Comment:
The core post meta bindings are now also editable inline in the editor which may be nice to add :)

Reply:
I'm not familiar with that, can you expand upon this or perhaps share a screenshot?

Reply:
In WordPress 6.6 it is now possible to actually edit the value that is saved in post meta right inline as you are editing the text of a bound block. So it no longer just displays the text but now also allows you to update it.

Reply:
I'm probably missing something, but I don't see how this is possible within a Block Bound pattern?  The only option available for editing is to "Edit Original", none of the innerblocks are selectable.  

![Screenshot 2024-10-01 at 8 46 35 AM](https://github.com/user-attachments/assets/80ea6193-7142-4527-a066-7501f9617141)

Reply:
It might only be possible to edit bound fields that are not inside of synced patterns. 

We can leave it out for now :)